war/src/main/js/pages/project/builds-card.js

```diff
@@ -0,0 +1,132 @@
+import debounce from "lodash/debounce";
+import behaviorShim from "@/util/behavior-shim";
+
+// Card/item controls
+const buildHistoryPage = document.getElementById("buildHistoryPage");
+const pageSearch = buildHistoryPage.querySelector(".jenkins-search");
+const pageSearchInput = buildHistoryPage.querySelector("input");
+const ajaxUrl = buildHistoryPage.getAttribute("page-ajax");
+const card = document.querySelector("#jenkins-builds");
+const contents = card.querySelector("#jenkins-build-history");
+const container = card.querySelector(".app-builds-container");
+const noBuilds = card.querySelector("#no-builds");
+
+// Pagination controls
+const paginationControls = document.querySelector("#controls");
+const paginationPrevious = document.querySelector("#up");
+const paginationNext = document.querySelector("#down");
+
+// Refresh variables
+let buildRefreshTimeout;
+const updateBuildsRefreshInterval = 5000;
+
+/**
+ * Refresh the 'Builds' card
+ * @param {QueryParameters}  options
+ */
+function load(options = {}) {
+  /** @type {QueryParameters} */
+  const params = Object.assign({}, options, { search: pageSearchInput.value });
```

Comment:
Updates from filtering seem really slow?

I checked on ci.jenkins.io on e.g. the Jenkins core builds and it was pretty much instant.
Whereas on my freestyle project with ~25 builds, I used a stopwatch and timed it to about 3.5 seconds